src/fsharp/TypeChecker.fs

```diff
@@ -14677,12 +14686,20 @@ module TcExceptionDeclarations =
         CheckForDuplicateConcreteType env id.idText id.idRange
         NewExn cpath id vis (TExnFresh (MakeRecdFieldsTable [])) attrs (doc.ToXmlDoc())
 
+    let mkName =
```

Comment:
Can you put this in "PrettyNaming".  Also I think there is a global string interning table around (though I didn't spot it in lib.fs or illib.fs), we could just use that here

Reply:
These generated names are currently being taken for each field even if not used. I think it could be better to have an instant array/list lookup than using the global idents table.